### PR TITLE
Adding new feature to add the teams while create the assembly via boo:PLO…

### PR DESCRIPTION
**Pull request by s0i00bu**
Adding new feature to add the teams while create the assembly via boo: PLONEOPS-19066

### CODE REVIEW

**Comment by s0i00bu** [comment]
This PR is not required.

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 1]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+	 * @param orgName      {mandatory}
+	 * @param assemblyName {mandatory}
+	 * @param teams        {mandatory}
+	 * @return teamList
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<CiResource> addTeamsByAssembly(String orgName, String assemblyName, List<String> teams) throws OneOpsClientAPIException {
+		List<CiResource> teamResourceList = new ArrayList<>();
+		List<Team> inputTeamList = new ArrayList<>();
+		if (!teams.isEmpty()){
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan I made this changes and push the changes to new PR
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 2]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+			for (String team: teams){
+				Team inputTeam = new Team();
+				inputTeam.setName(team);
+				inputTeamList.add(inputTeam);
+			}
+		}
+		List<Team> teamResponseList = listOrganizationTeams(orgName);
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 3]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+		List<Team> teamsList = new ArrayList<>();
+		if (teamResponseList != null && !teamResponseList.isEmpty() && !inputTeamList.isEmpty()) {
+			for (Team teamResponse : teamResponseList) {
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 4]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+				for (Team inputTeam: inputTeamList){
+					if (teamResponse.getName().equalsIgnoreCase(inputTeam.getName())){
+						Team team = new Team();
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 5]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+	/**
+	 * List of teams for the given organization
+	 *
+	 * @param orgName {mandatory}
+	 * @return
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<Team> listOrganizationTeams(String orgName) throws OneOpsClientAPIException {
+		if(orgName == null || orgName.length() == 0) {
+			String msg = "Missing assembly name to fetch details";
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 6]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+			throw new OneOpsClientAPIException(msg);
+		}
+		RequestSpecification request = createRequest();
+		Response response = request.get(IConstants.ORGANIZATION_URI + IConstants.TEAMS);
+		if(response != null) {
+			if(response.getStatusCode() == 200 || response.getStatusCode() == 302) {
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 7]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+						team.setName(teamResponse.getName());
+						teamsList.add(team);
+					}
+				}
+			}
+		}
+		if (!teamsList.isEmpty()) {
+			for (Team team : teamsList) {
+				if (null != team.getId()) {
+					List<CiResource> ciResourceList = addingTeamNameByAssembly(orgName, assemblyName, team.getId());
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 8]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+	/**
+	 * Adding Team Name By Assembly
+	 *
+	 * @param orgName {mandatory}
+	 * @param assemblyName {mandatory}
+	 * @param teamId {mandatory}
+	 * @return
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<CiResource> addingTeamNameByAssembly(String orgName, String assemblyName, String teamId) throws OneOpsClientAPIException {
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

File: src/main/java/com/oneops/api/resource/Assembly.java [hunk 9]
```diff
@@ -312,4 +310,123 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
...
+		List<Team> assemblyTeams =listAssemblyTeams(assemblyName);
```
[the remark this answers is not in the document]
Comment (s0i00bu) [reply src/main/java/com/oneops/api/resource/Assembly.java]: @aneeshpadmanabhan  I made this changes and push the changes to new PR.
https://github.com/oneops/OneOpsAPIClient/pull/51

**Comment by aneeshpadmanabhan** [comment]
These changes were merged as part of https://github.com/oneops/OneOpsAPIClient/pull/51

**Pull request closed without merging**